### PR TITLE
Optimizing web load speed 🚀 

### PR DESCRIPTION
**Pull request by sanjaiyan-dev**
[Deferring script load for faster web page load](https://web.dev/efficiently-load-third-party-javascript/#defer).

_Sorry if I made any mistakes :(_

### CODE REVIEW

**Comment by domesticmouse** [comment]
PTAL @khanhnwin

**Comment by atsansone** [comment]
@parlough , @khanhnwin : I see no issues caused by this change (I staged it locally), but I'm not sure what need this fulfills. What do you think?

File: src/_layouts/base.html [hunk 1]
```diff
@@ -67,6 +67,24 @@
...
+    <script defer src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap/4.6.2/js/bootstrap.min.js" integrity="sha512-7rusk8kGPFynZWu26OKbTeI+QPoYchtxsmPeBqkHIEXJxeun4yJ4ISYe7C6sz9wdxeE1Gk3VxsIWgCZTc+vX3g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
+    <script defer src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
+    <script defer src="https://cdnjs.cloudflare.com/ajax/libs/js-cookie/3.0.1/js.cookie.min.js" integrity="sha512-wT7uPE7tOP6w4o28u1DN775jYjHQApdBnib5Pho4RB0Pgd9y7eSkAV1BTqQydupYDB9GBhTcQQzyNMPMV3cAew==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
+
+    <script defer src="{{ '/assets/js/vendor/code-prettify/lang-css.js' | append: cache_bust }}"></script>
+    <script defer src="{{ '/assets/js/vendor/code-prettify/prettify.js' | append: cache_bust }}"></script>
```
Comment (parlough) [inline src/_layouts/base.html:77]: The prettify script needs to be included first:

```suggestion
    <script defer src="{{ '/assets/js/vendor/code-prettify/prettify.js' | append: cache_bust }}"></script>
    <script defer src="{{ '/assets/js/vendor/code-prettify/lang-css.js' | append: cache_bust }}"></script>
```

File: src/_layouts/base.html [hunk 2]
```diff
@@ -67,6 +67,24 @@
...
+    <script defer src="{{ '/assets/js/vendor/code-prettify/lang-dart.js' | append: cache_bust }}"></script>
+    <script defer src="{{ '/assets/js/vendor/code-prettify/lang-yaml.js' | append: cache_bust }}"></script>
+    <script defer src="{{ '/assets/js/tabs.js' | append: cache_bust }}"></script>
+    <script defer src="{{ '/assets/js/archive.js' | append: cache_bust }}"></script>
+    <script defer src="{{ '/assets/js/main.js' | append: cache_bust }}"></script>
+    
+    {% for script in page.js %}
+      <script defer src="{{ script.url | default: js }}" {% if script.defer %}defer{% endif %}></script>
+    {% endfor %}
```
Comment (parlough) [inline src/_layouts/base.html:86]: This already has logic for optionally deferring with the `{% if script.defer %}defer{% endif %}`. We don't want to force it.

```suggestion
    {% for script in page.js %}
      <script src="{{ script.url | default: js }}" {% if script.defer %}defer{% endif %}></script>
    {% endfor %}
```

**Review by parlough — CHANGES_REQUESTED** [review]
Thanks for the PR. Some initial requests.

**Comment by domesticmouse** [comment]
LGTM once @parlough's changes are adopted

**Comment by sanjaiyan-dev** [comment]
> @parlough , @khanhnwin : I see no issues caused by this change (I staged it locally), but I'm not sure what need this fulfills. What do you think?

Can you give me a small amount of time please there are some more changes to be done.

Extremely sorry for the inconvenience

File: src/_layouts/base.html (hunk 2 above)
Comment (sanjaiyan-dev) [reply src/_layouts/base.html]: > This already has logic for optionally deferring with the `{% if script.defer %}defer{% endif %}`. We don't want to force it.

@parlough  Done 🤝

> Preloaded those script and removed defer

**Comment by sanjaiyan-dev** [comment]
Done ✨

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
PTAL @khanhnwin

**Comment by sfshaza2** [comment]
Is this ready to land? It's not entirely clear, @domesticmouse.

**Comment by domesticmouse** [comment]
I'd like @khanhnwin to take a look.

**Comment by khanhnwin** [comment]
Apologies for being late to review! LGTM! Thanks @sanjaiyan-dev & everyone!

**Review by khanhnwin — APPROVED** [review]

**Pull request merged**